erts/emulator/beam/big.h

```diff
@@ -120,9 +120,12 @@ typedef Uint  dsize_t;	 /* Vector size type */
 #endif
 
 int big_decimal_estimate(Wterm);
+int big_integer_estimate(Wterm, Uint);
 Eterm erts_big_to_list(Eterm, Eterm**);
+Eterm erts_big_to_list_by_base(Eterm, Eterm**, Uint);
 char *erts_big_to_string(Wterm x, char *buf, Uint buf_sz);
 Uint erts_big_to_binary_bytes(Eterm x, char *buf, Uint buf_sz);
+Uint erts_big_to_binary_bytes_by_base(Eterm x, char *buf, Uint buf_sz, Uint base);
```

Comment:
Consider turning the decimal-specific functions/BIFs into wrappers around the by-base functions, it'll be easier to maintain.

Reply:
I turned the basic functions into wrappers. Do I have to rework BIFs completely or this is ok?

Reply:
I think that it makes sense to rework the bifs so that they can share code. Possibly it would be a good thing to specialize `integer_to_list/1` (and `to_binary/1`) of small integers as that will be the most common case and is used quite a lot.